diff --git a/engines.py b/engines.py
index 8a729c3..08559c4 100644
--- a/engines.py
+++ b/engines.py
@@ -2,3 +2,4 @@ from talon import Context, speech_system
 from talon.engines.webspeech import WebSpeechEngine
 webspeech = WebSpeechEngine()
 speech_system.add_engine(webspeech)
+
diff --git a/languages.py b/languages.py
index 055db98..85c404a 100644
--- a/languages.py
+++ b/languages.py
@@ -6,3 +6,4 @@ mod.mode("portuguese")
 mod.mode("russian")
 mod.mode("spanish")
 mod.mode("swedish")
+mod.mode("dutch")
